packages/devtools_app/lib/src/shared/memory/class_name.dart

```diff
@@ -102,15 +101,16 @@ class HeapClassName {
   final String className;
   final String library;
 
-  String get fullName => library.isNotEmpty ? '$library/$shortName' : shortName;
+  late final String fullName =
+      library.isNotEmpty ? '$library/$shortName' : shortName;
 
-  bool get isSentinel => className == 'Sentinel' && library.isEmpty;
+  late final isSentinel = className == 'Sentinel' && library.isEmpty;
 
-  bool get isRoot => className == 'Root' && library.isEmpty;
+  late final isRoot = className == 'Root' && library.isEmpty;
 
   /// Whether a class can hold a reference to an object
   /// without preventing garbage collection.
-  bool get isWeakEntry {
+  late final bool isWeakEntry = () {
```

Comment:
Is this just to lazily initialize `isWeakEntry`? This behavior might not be obvious to everyone, so I'd recommend leaving `isWeakEntry` as a getter and creating a nullable private property that is set the first time `isWeakEntry` is invoked.

We need to do extra runtime checks with late fields anyway, so there's no benefit doing it this way over using a nullable private field when it comes to performance.

Reply:
I started doing `late final` initializers after @jacob314 suggested it as a recommended way to do lazy initializations in one of code reviews. 

For people who know this pattern, `late final` seems to be more concise and readable lazy initialization than private field.

Jacob, could you comment?

Reply:
In general, I think it is reasonable to expect everyone to understand what `late final` means in Dart as it is a feature that was thoughtfully added to Dart as part of supporting null safety. I do agree it is a bit of surprising feature at first.
There is a tiny performance benefit as compilers can know that a late final field is only initialized once. This is also why we can't have `late` objects that have a `dispose` lifecycle because that would violate the compilers assumptions.

I wonder if makes sense to get some guidance on this added to https://dart.dev/guides/language/effective-dart

Reply:
AFAIK, each access to a `late final` field still requires a sentinel check to ensure the field has been initialized, which is basically the equivalent to a null check on each access, so there's no benefit over using one or the other performance wise.

I do personally think this is less readable (declaring a closure only to invoke it immediately looks smelly to me) than the alternative.

Reply:
At the very least, I think these closures should be made into named methods to avoid having large blocks of initialization logic inline.

Reply:
Separated calculations into method.